asapdiscovery-data/asapdiscovery/data/state_expanders/expansion_tag.py

```diff
@@ -0,0 +1,34 @@
+import uuid
+
+from pydantic import UUID4, BaseModel, Field
+
+
+class StateExpansionTag(BaseModel):
+    id: UUID4 = Field(uuid.uuid4(), description="UUID for this molecule")
```

Comment:
I like the idea of keeping track of the relationship between the parent and child. But I think we should use identifiers tied to the molecule's identity like the inchi/inchikey as this will make it much easier to workout the parent and the id should not change depending on when the workflow was run.

Reply:
Yeah ok that makes sense, should be an actual property.

Reply:
Now uses INCHIKey